src/services/healthcareProfessionalService.ts

```diff
@@ -55,51 +55,67 @@ export async function addHealthcareProfessional(
     }
 }
 
-export async function addHealthcareProfessionalToFacility(facilityId: string, 
-    healthcareProfessional: gqlTypes.HealthcareProfessionalInput) {
+/**
+ * Creates a HealthcareProfessional and adds it to the listed facilities
+ * @param healthcareProfessionalInput 
+ * @returns A HealthcareProfessional object
+ */
+export async function addHealthcareProfessionalToFacility( 
+    healthcareProfessionalInput: gqlTypes.HealthcareProfessionalInput
+) {
+    const addHealthcareProfessionalResult : Result<dbSchema.HealthcareProfessional> = {
+        hasErrors: false,
+        errors: []
+    }
+
+    if (healthcareProfessionalInput.facilityIds.length == 0) {
+        throw CustomErrors.missingInput('The list of facilityIds cannot be empty.')
```

Comment:
we should migrate off the custom errors to using the result object with the error code. This will let us have localized language error messages. 
could add an error code like `ADDHEALTHCAREPROF_FACILITYIDS_REQUIRED`